lib/Frontend/ParseableInterfaceModuleLoader.cpp

```diff
@@ -713,9 +712,6 @@ class ParseableInterfaceModuleLoaderImpl {
     if (validationInfo.status != serialization::Status::Valid)
       return false;
 
-    assert(validationInfo.name == moduleName &&
```

Comment:
Any reason this was removed as part of these changes? I think it's fine to remove.

Reply:
The change uncovered a bug: now that we use this function for adjacent swiftmodules as well as cached swiftmodules, the assertion is no longer valid. This was tested by test/Serialization/load-wrong-name.swift.